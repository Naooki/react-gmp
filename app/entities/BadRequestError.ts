export default interface BadRequestError {
  messages: string[];
}
