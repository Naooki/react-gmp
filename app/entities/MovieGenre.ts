enum MovieGenre {
  Crime = 'Crime',
  Documentary = 'Documentary',
  Horror = 'Horror',
  Drama = 'Drama',
  Romance = 'Romance',
  Animation = 'Animation',
  Adventure = 'Adventure',
  Family = 'Family',
  Comedy = 'Comedy',
  Fantasy = 'Fantasy',
  ScienceFiction = 'Science Fiction',
  Action = 'Action',
}

export default MovieGenre;
