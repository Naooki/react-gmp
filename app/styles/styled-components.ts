import * as styledComponents from 'styled-components';

// theme.ts
// your theme variables
export interface IThemeInterface {
  text: string;
  primary: string;
  componentBackground: string;
  componentBackgroundSecondary: string;
  footerBackground: string;
  pageBackground: string;
}

export const theme = {
  default: {
    text: '#fff',
    primary: '#f65261',
    componentBackground: '#232323',
    componentBackgroundSecondary: 'rgba(85,85,85,0.8)',
    footerBackground: '#424242',
    pageBackground: '#555',
  },
};
const {
  default: styled,
  css,
  createGlobalStyle,
  keyframes,
  ThemeProvider,
} = styledComponents as styledComponents.ThemedStyledComponentsModule<
  IThemeInterface
>;

export { css, createGlobalStyle, keyframes, ThemeProvider };
export default styled;
