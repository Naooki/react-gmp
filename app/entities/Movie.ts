import MovieGenre from './MovieGenre';

export interface Movie {
  id: string;
  releaseDate: string;
  about: string;
  title: string;
  imageUrl?: string;
  duration: number;
  rating: number;
  genres: MovieGenre[];
}
