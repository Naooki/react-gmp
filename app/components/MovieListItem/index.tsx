import * as React from 'react';

import { Movie } from 'entities/Movie';

interface Props {
  movie: Movie;
}

const MovieListItem = (props: Props) => {
  const releaseYear = props.movie.releaseDate;

  return (
    <li>
      <img src={props.movie.imageUrl} alt="movie poster" />
      <div className="movie-description">
        <h3>{props.movie.title}</h3>
        <div className="movie-year">{releaseYear}</div>
        <span className="movie-about">{props.movie.about}</span>
      </div>
    </li>
  );
};

export default MovieListItem;
