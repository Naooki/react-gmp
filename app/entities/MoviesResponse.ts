import { Movie } from './Movie';

export default interface MoviesResponse {
  data: Movie[];
  total: number;
  offset: number;
  limit: number;
}
