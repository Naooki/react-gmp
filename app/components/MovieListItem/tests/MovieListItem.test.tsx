import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render, cleanup, fireEvent } from '@testing-library/react';

import { Movie } from 'entities/Movie';
import MovieGenre from 'entities/MovieGenre';
import { theme } from 'styles/styled-components';
import MovieListItem from '..';

let props;

const movie: Movie = {
  title: 'La La Land',
  tagline: "Here's to the fools who dream.",
  vote_average: 7.9,
  vote_count: 6782,
  release_date: '2016-12-29',
  poster_path:
    'https://image.tmdb.org/t/p/w500/ylXCdC106IKiarftHkcacasaAcb.jpg',
  overview:
    'Mia, an aspiring actress, serves lattes to movie stars in between auditions and Sebastian, a jazz musician, scrapes by playing cocktail party gigs in dingy bars, but as success mounts they are faced with decisions that begin to fray the fragile fabric of their love affair, and the dreams they worked so hard to maintain in each other threaten to rip them apart.',
  budget: 30000000,
  revenue: 445435700,
  runtime: 128,
  genres: [MovieGenre.Comedy, MovieGenre.Drama, MovieGenre.Romance],
  id: 313369,
};

beforeEach(() => {
  props = {
    movie,
    onMovieEdit: jest.fn(),
    onMovieDelete: jest.fn(),
    onMovieClick: jest.fn(),
  };
});

afterEach(cleanup);

describe('<MovieListItem />', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme.default}>
        <MovieListItem {...props} />
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onMovieClick on image click', () => {
    const { getByAltText } = render(
      <ThemeProvider theme={theme.default}>
        <MovieListItem {...props} />
      </ThemeProvider>,
    );

    expect(props.onMovieClick).not.toHaveBeenCalled();
    fireEvent(
      getByAltText('Movie poster'),
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      }),
    );
    expect(props.onMovieClick).toHaveBeenCalledWith(313369);
  });
});
