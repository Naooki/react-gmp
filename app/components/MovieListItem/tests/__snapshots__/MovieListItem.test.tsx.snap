// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieListItem /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxmLO kxxPnf"
  >
    <div
      class="sc-AxjAm iFRHBp movie-item-menu "
    >
      <button
        class="sc-AxirZ sTVkX"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-ellipsis-v fa-w-6 "
          data-icon="ellipsis-v"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <img
      alt="Movie poster"
      class="sc-Axmtr jrCDfN"
      src="https://image.tmdb.org/t/p/w500/ylXCdC106IKiarftHkcacasaAcb.jpg"
    />
    <div
      class="sc-fzozJi knnLCK"
    >
      <h3
        class="sc-fzoLsD gsLTYY"
      >
        La La Land
      </h3>
      <div
        class="sc-fzpans epIulx"
      >
        2016-12-29
      </div>
      <span
        class="sc-fzplWN iCUPXI"
      >
        Comedy, Drama, Romance
      </span>
    </div>
  </div>
</DocumentFragment>
`;
